newsroom/utils.py

```diff
@@ -149,27 +152,59 @@ async def get_json_or_400_async(req: Request):
     return data
 
 
-def success_response(data: Any, headers: Sequence = ()):
+def success_response(data: Any, headers: Sequence = (), status_code: int = 200):
     """
     Shortcut to return a `superdesk.core.web.Response` with a
     status_code 200 and empty headers by default
     """
     if isinstance(data, BaseModel):
         data = data.model_dump(by_alias=True, exclude_unset=True)
 
-    return Response(data, 200, headers)
+    return Response(data, status_code, headers)
 
 
-def response_from_validation(error: ValidationError, code=400):
+def parse_validation_error(error: ValidationError) -> dict[str, str]:
```

Comment:
Same thing here, we can maybe check `isinstance(response.body, ValidationError)` in [superdesk-core -> app](https://github.com/superdesk/superdesk-core/blob/async/superdesk/factory/app.py#L259). This way we can do the following using this functionality from the core framework:

```python
try:
    ids = await service.create([model_data])
    return ids[0]
except ValidationError as err:
    return Response(err, 400)
```

With this supported as a core feature, we won't need `create_or_abort` either, as a Newshub only feature

Reply:
You make a good point. An improvement to your suggestion, what if we use Flask/Quart's register_error_handler functionality like we're already doing for [SuperdeskApiErrors](https://github.com/superdesk/newsroom-core/blob/async/newsroom/factory/app.py#L210).

This way we can customise the response for Newshub and Superdesk (as I believe they use different formats for the validation errors to the client), and not need any request specific helpers (create, update and delete validation errors would be caught by the same error handler).

Reply:
yes, this is great idea. I'll try that and update the PRs.

Reply:
done.